faculty/session.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2018-2019 ASI Data Science
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from datetime import datetime, timedelta
+from collections import namedtuple
+
+import requests
+import pytz
+from six.moves import urllib
+
+import faculty.config
+
+
+AccessToken = namedtuple("AccessToken", ["token", "expires_at"])
+
+
+class MemoryAccessTokenCache(object):
+    def __init__(self):
+        self._store = {}
+
+    def get(self, profile):
+        access_token = self._store.get(profile)
+        utc_now = datetime.now(tz=pytz.utc)
+        if access_token is None or access_token.expires_at < utc_now:
+            return None
+        else:
+            return access_token
+
+    def add(self, profile, access_token):
+        self._store[profile] = access_token
+
+
+def _service_url(profile, service, endpoint=""):
```

Comment:
`_service_url` and `_get_access_token` are only used AFAICT from within `Session`. Is there a reason not to put them as static methods of the class?

Reply:
I generally prefer putting less stuff in the class as it's easier to reason about dependencies and thinking about tests when they're obviously not dependent on the class itself.